src/common/utils.py

```diff
@@ -389,3 +401,31 @@ def retrieve_max_cluster_size(region, proxy_config, asg_name, fallback):
         )
         log.critical(error_msg)
         raise CriticalError(error_msg)
+
+
+def get_cluster_instance_info(stack_name, region, proxy_config, include_master=False):
+    """Return a list of instance_id that are in the cluster."""
+    try:
+        instances_in_cluster = []
+        ec2_client = boto3.client("ec2", region_name=region, config=proxy_config)
+        instance_paginator = ec2_client.get_paginator("describe_instances")
+        nodes_to_include = ["Compute", "Master"] if include_master else ["Compute"]
+
+        for page in instance_paginator.paginate(
+            Filters=[
+                {"Name": "tag:Application", "Values": [stack_name]},
+                {"Name": "tag:Name", "Values": nodes_to_include},
+            ]
+        ):
+            for reservation in page.get("Reservations"):
+                for instance in reservation.get("Instances"):
+                    is_alive = instance.get("State").get("Name") in INSTANCE_ALIVE_STATES
+                    instance_id = instance.get("InstanceId")
+                    if is_alive:
+                        instances_in_cluster.append(instance_id)
+
+        return instances_in_cluster
+
+    except Exception as e:
+        logging.error("Failed retrieving instance_ids for cluster %s with exception: %s", stack_name, e)
+        raise
```

Comment:
is the sqswatcher crashing if this fails? if so we should handle failures in handling such messages with the same requeuing mechanism we use for the other events

Reply:
`sqswatcher` will handle this in `_process_scheduled_maintenance_event`